File: merge-two-sorted-lists/merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        
        ListNode* l3=new ListNode();
        ListNode* head=l3;
        
        while(l1!=NULL and l2!=NULL){
            
            if(l2->val>l1->val){
                l3->next=l1;
                l1=l1->next;
            }
            
            else{
                l3->next=l2;
                l2=l2->next;
            } 
            
        l3=l3->next;
        }
        if(l1!=NULL)    l3->next=l1;
        
        else if(l2!=NULL) l3->next=l2;
        
        
       return head->next; 
    
    }
};

File: merge-two-sorted-lists/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        
        dummy=ListNode(0)
    
        l3=dummy
            
        while(l1 and l2):   #Checking neither is none
            
            if(l1.val<=l2.val):
                
                l3.next=l1
                l1=l1.next
                
            else:
                l3.next=l2
                l2=l2.next
            
            l3=l3.next
            
            
        if(l1 is None):
            l3.next=l2
        if(l2 is None):
            l3.next=l1
                
        return dummy.next
                
            
        